pkgs/development/python-modules/cartopy/default.nix

```diff
@@ -0,0 +1,38 @@
+{ buildPythonPackage, lib, fetchPypi,
+  pytest, filelock, mock, pep8,
+  cython,
+  six, pyshp, shapely, geos, proj, numpy,
+  gdal, pillow, matplotlib, pyepsg, pykdtree, scipy, owslib, fiona
+}:
+
+buildPythonPackage rec {
+  pname = "Cartopy";
+  version = "0.17.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0q9ckfi37cxj7jwnqnzij62vwcf4krccx576vv5lhvpgvplxjjs2";
+  };
+
+  checkInputs = [ filelock mock pytest pep8  ];
+#  doCheck = false;
+
+  buildInputs = [ cython ];
+
+  propagatedBuildInputs = [
+    # required
+    six pyshp shapely geos proj numpy
+
+    #optional
+    gdal pillow matplotlib pyepsg pykdtree scipy owslib fiona
+  ];
+
+  meta = with lib; {
+    description = ''Python package designed for geospatial data processing in
```

Comment:
This description can be shortened for brevity.

Something like

"Process geospatial data to create maps and perform analyses"